build.gradle

```diff
@@ -6,6 +6,9 @@ plugins {
     id 'net.ltgt.errorprone' version '3.1.0'
 }
 
+group 'io.github.eisopux'
+version '0.2.0'
+
```

Comment:
Thanks, looks good, will this downloader be used as a dependency in the future?

Reply:
Also, do you need to keep all version numbers consistent? Or just different numbers for different repos?

Reply:
> Thanks, looks good, will this downloader be used as a dependency in the future?

What do you mean with "this downloader" and "as a dependency"?

Reply:
> Also, do you need to keep all version numbers consistent? Or just different numbers for different repos?

I was thinking whether we should use one version for all three projects, but there is no need to keep them in sync - `cf-vscode` uses the `cf-ls-downloader` to download the latest release of `eisop/cf` and `cf-ls`. The `cf-ls-downloader` likely won't need many updates. The `cf-ls` and the `cf-vscode` projects hopefully can evolve separately and, ideally, we'll have other IDEs that use the same LSP also. So I think having separate versions is OK.
What do you think?

Reply:
Ok, I agree now. Each extension should have a separate version number for developing. And if we think the downloader is stable, we can make a 1.0.0 release.

Reply:
> > Thanks, looks good, will this downloader be used as a dependency in the future?
> 
> What do you mean with "this downloader" and "as a dependency"?

I am just wondering why adding metadata in the project.

Reply:
> I am just wondering why adding metadata in the project.

In general, I would say it's good to put this metadata into your build files.
Here, it results in the built jar file being named `./build/libs/checker-framework-languageserver-downloader-0.3.0-all.jar` instead of `./build/libs/checker-framework-languageserver-downloader-all.jar`.
So the build artifact also nicely contains the version number.
In the releases, we need to make sure that the first artifact is the built jar file, because that's an assumption of the cf-ls-downloader (we should document that better).
Is there any downside in including this metadata in the build script?